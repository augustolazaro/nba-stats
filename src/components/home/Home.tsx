import * as React from 'react'
import graphql from 'graphql-tag'
import styled from 'styled-components'

import { formatDate } from '../utils'

import TeamLogo from '../common/TeamLogo'
//@ts-ignore
import DateSelector from '../common/DateSelector'

import createQueryRenderer from '../../hocs/createQueryRenderer'

const Container = styled.div`
  display: flex;
  flex-direction: column;
`

const Content = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`

const GameCard = styled.div`
  display: flex;
  flex-direction: column;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 20px;
  margin: 10px;
  background-color: white;
  width: 290px;
`

const CardHeader = styled.div`
  display: flex;
  align-items: center;
`

const ScoreWrapper = styled.div.attrs<{ home: string }>({})`
  flex: 1;
  display: flex;
  align-items: center;
`

const TeamAbbr = styled.span`
  font-size: 18px;
  font-weight: bold;
`

const Score = styled.span`
  font-size: 20px;
  margin: 0 5px;
`

const Divider = styled.span`
  margin: 0 10px;
  font-size: 12px;
`

const HelperText = styled.span`
  font-size: 13px;
  text-align: center;
`

const EmptyMessage = styled.span`
  font-size: 16px;
`

type Props = {
  query: {
    games: Array<{}>,
  },
  refetch: (args: object) => void,
}

const Home = ({ query, refetch }: Props) => {
  const { games } = query 

  const handleDateChange = (date: Date) => {
    refetch({
      variables: {
        date,
      },
      updateQuery: (prev: any, { fetchMoreResult }: any) => {
        return {
          ...prev, 
          games: fetchMoreResult.games,
        }
      },
    })
  }

  return (
    <Container>
      <DateSelector onChange={handleDateChange} />
      <Content>
        {!games.length && (<EmptyMessage className='empty-message'>No scheduled games on this day</EmptyMessage>)}
        {!!games.length && games.map((game: any) => (
          <GameCard>
            <HelperText>{formatDate(game.date)}</HelperText>
            <CardHeader>
              <ScoreWrapper style={{ justifyContent: 'flex-end' }}>
                <TeamAbbr>{game.visitorTeam && game.visitorTeam.abbreviation}</TeamAbbr>
                <TeamLogo src={game.visitorTeam && game.visitorTeam.logo}></TeamLogo>
                <Score>{game.visitorTeamScore}</Score>
              </ScoreWrapper>
              <Divider>x</Divider>
              <ScoreWrapper>
                <Score>{game.homeTeamScore}</Score>
                <TeamLogo src={game.homeTeam && game.homeTeam.logo}></TeamLogo>
                <TeamAbbr>{game.homeTeam && game.homeTeam.abbreviation}</TeamAbbr>
              </ScoreWrapper>
            </CardHeader>
            <HelperText>{game.status}</HelperText>
          </GameCard>
        ))}
      </Content>
    </Container>
  )
}

export default createQueryRenderer(Home, {
  query: graphql`
    query HomeQuery ($date: String) {
      games(date: $date) {
        date
        status
        homeTeamScore
        visitorTeamScore
        homeTeam {
          logo
          abbreviation
        }
        visitorTeam {
          logo
          abbreviation
        }
      }
    }
  `,
})