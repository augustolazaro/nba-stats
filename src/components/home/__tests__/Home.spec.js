import * as React from 'react'
import { MockedProvider } from 'react-apollo/test-utils';
import graphql from 'graphql-tag';
import * as renderer from 'react-test-renderer';
import wait from 'waait'

import Home from '../Home'
import { ThemeProvider } from 'styled-components';
import { theme } from '../../../index'

const mock = [
  {
    request: {
      query: graphql`
        query HomeQuery ($date: String) {
          games(date: $date) {
            date
            status
            homeTeamScore
            visitorTeamScore
            homeTeam {
              logo
              abbreviation
            }
            visitorTeam {
              logo
              abbreviation
            }
          }
        }
      `,
    },
    result: {
      data: {
        games: [],
      }
    },
  }
]

it('should render teams list with no errors', async () => {
  const component = renderer.create(
    <MockedProvider mocks={mock} addTypename={false}>
      <ThemeProvider theme={theme}>
        <Home theme={theme} />
      </ThemeProvider>
    </MockedProvider>
  );

  await wait(0)

  // expect(component.root.findByProps({ className: 'empty-message'}).children).toBeTruthy()
  expect(component.toJSON()).toMatchSnapshot()
})