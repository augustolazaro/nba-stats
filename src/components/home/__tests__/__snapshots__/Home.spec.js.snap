// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render teams list with no errors 1`] = `
<div
  className="sc-bZQynM AMncx"
>
  <div
    className="sc-bwzfXH jzJyNX"
  >
    <div
      className="sc-bxivhb ksDfcC"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height={50}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width={50}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="15 18 9 12 15 6"
        />
      </svg>
    </div>
    <span
      className="sc-htpNat fIjutD"
    >
      Wednesday, 20/February
    </span>
    <div
      className="sc-bxivhb ksDfcC"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height={50}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width={50}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="9 18 15 12 9 6"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-gzVnrw jXalhU"
  >
    <span
      className="empty-message sc-fjdhpX gelbAW"
    >
      No scheduled games on this day
    </span>
  </div>
</div>
`;
